src/slurmrestd_ops.py

```diff
@@ -0,0 +1,333 @@
+            elif action == "stop":
+                systemd.service_stop(service)
+            elif action == "restart":
+                systemd.service_restart(service)
+            else:
+                logger.debug("Only start, stop, restart supported.")
+        except SlurmrestdException("Cannot restart slurmrestd.") as e:  # type: ignore [misc]
+            logger.error(e)
+            return False
+        return True
+
+    def _create_slurmrestd_user_group(self) -> None:
+        """Create the slurmrestd user."""
+        logger.info("#### Creating slurmrestd user and group")
+
+        try:
+            subprocess.check_output(
+                ["groupadd", "--gid", self._slurmrestd_group_id, self._slurmrestd_group_name]
+            )
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 9:
+                logger.warning("## Group already exists.")
+            else:
+                logger.error(f"## Error creating group: {e}")
+                raise (e)
+
+        try:
+            subprocess.check_output(
+                [
+                    "adduser",
+                    "--system",
+                    "--gid",
+                    self._slurmrestd_group_id,
+                    "--uid",
+                    self._slurmrestd_user_id,
+                    "--no-create-home",
+                    "--home",
+                    "/nonexistent",
+                    self._slurmrestd_user_name,
+                ]
+            )
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 9:
+                logger.warning("## User already exists.")
+            else:
+                logger.error(f"## Error creating user: {e}")
+                raise (e)
+
+        logger.info("#### Created slurmrestd user and group")
+
+    @property
+    def slurm_conf_path(self) -> Path:
+        """Return the slurm conf path."""
+        return self._slurm_conf_path
+
+    @property
+    def _munge_key_path(self) -> Path:
+        """Return the full path to the munge key."""
+        return Path("/etc/munge/munge.key")
+
+    @property
+    def _munged_systemd_service(self) -> str:
+        """Return the name of the Munge Systemd unit file."""
+        return "munge.service"
+
+    @property
+    def _slurmrestd_systemd_service(self) -> str:
+        """Return the Slurm systemd unit file."""
+        return "slurmrestd.service"
+
+    @property
+    def _slurmrestd_user_id(self) -> str:
+        """Return the slurmrestd user ID."""
+        return "64031"
+
+    @property
+    def _slurmrestd_user_name(self) -> str:
+        """Return the slurmrestd user."""
+        return "slurmrestd"
+
+    @property
+    def _slurmrestd_group_id(self) -> str:
+        """Return the slurmrestd group ID."""
+        return "64031"
+
+    @property
+    def _slurmrestd_group_name(self) -> str:
+        """Return the slurmrestd group."""
+        return "slurmrestd"
```

Comment:
Would it possibly be better to have these set as constants within a `constants.py` module rather than have them be object properties? Since this isn't a public package, I'm not worried about folks potentially overwriting the values.